sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormTrainingClient/FormTrainingClientMockTests.cs

```diff
@@ -69,6 +69,41 @@ public async Task StartTrainingEncodesBlankSpaces()
             }
         }
 
+        [Test]
+        public async Task GetCustomModelDoesNotNeedTrainResult()
+        {
+            // When a model is not Ready, the "trainResult" property is not returned.
+            // We're mimicking this behavior here to make sure that we can parse the
+            // response even without the "trainResult" property.
+
+            // Ideally this test should be live, but the service behavior is
+            // non-deterministic. Any delay in sending the request could give the
+            // model enough time to change its status to Ready.
```

Comment:
I would really love to have this one as a live test, but decided to keep it local to avoid flakiness.